src/NLog/Targets/Wrappers/LimitingTargetWrapper.cs

```diff
@@ -0,0 +1,173 @@
+        /// </summary>
+        /// <remarks>
+        /// Messages received after <see cref="MessageLimit"/> has been reached in the current <see cref="Interval"/> will be discarded.
+        /// </remarks>
+        [DefaultValue(1000)]
+        public int MessageLimit { get; set; }
+
+        /// <summary>
+        /// Gets or sets the interval in which messages will be written up to the <see cref="MessageLimit"/> number of messages.
+        /// </summary>
+        /// <remarks>
+        /// Messages received after <see cref="MessageLimit"/> has been reached in the current <see cref="Interval"/> will be discarded.
+        /// </remarks>
+        [DefaultValue(typeof(TimeSpan), "01:00")]
+        public TimeSpan Interval { get; set; }
+
+        /// <summary>
+        /// Gets the <c>DateTime</c> when the current <see cref="Interval"/> will be reset.
+        /// </summary>
+        public DateTime IntervalResetsAt { get { return firstWriteInInterval + Interval; } }
+
+        /// <summary>
+        /// Gets the number of <see cref="AsyncLogEventInfo"/> written in the current <see cref="Interval"/>.
+        /// </summary>
+        public int MessagesWrittenCount { get; private set; }
+
+        /// <summary>
+        /// Initializes the target and resets the current Interval and <see cref="MessagesWrittenCount"/>.
+        ///  </summary>
+        protected override void InitializeTarget()
+        {
+            if(this.MessageLimit<=0)
+                throw new NLogConfigurationException("The LimitingTargetWrapper\'s MessageLimit property must be > 0.");
+            if(this.Interval<=TimeSpan.Zero)
+                throw new NLogConfigurationException("The LimitingTargetWrapper\'s property Interval must be > 0.");
+
+            base.InitializeTarget();
+            ResetInterval();
+            InternalLogger.Trace("LimitingTargetWraper '{0}': initialized with MessageLimit={1} and Interval={2}.", Name, MessageLimit, Interval);
+        }
+
+
+        /// <summary>
+        /// Writes log event to the wrapped target if the current <see cref="MessagesWrittenCount"/> is lower than <see cref="MessageLimit"/>.
+        /// If the <see cref="MessageLimit"/> is already reached, no log event will be written to the wrapped target.
+        /// <see cref="MessagesWrittenCount"/> resets when the current <see cref="Interval"/> is expired.
+        /// </summary>
+        /// <param name="logEvent">Log event to be written out.</param>
+        protected override void Write(AsyncLogEventInfo logEvent)
+        {
+            if (IsIntervalExpired())
+            {
+                ResetInterval();
+                InternalLogger.Debug("LimitingWrapper '{0}': new interval of '{1}' started.", Name, Interval);
+            }
+
+            if (MessagesWrittenCount < MessageLimit)
+            {
+                this.WrappedTarget.WriteAsyncLogEvent(logEvent);
+                MessagesWrittenCount++;
```

Comment:
Shoudn't this be something like `Interlocked.Increment(ref MessagesWrittenCount)` or something?